Add UPDATE support for Iceberg (#24281)
prestodb/presto: presto-docs/src/main/sphinx/connector/iceberg.rst

     comment   | varchar |       |
     (3 rows)

UPDATE
^^^^^^

> The iceberg connector supports `UPDATE` operations on Iceberg tables. See :doc:`../sql/update` for more details

Reviewer: ```suggestion
The Iceberg connector supports :doc:`../sql/update` operations on Iceberg tables. 
```

> on the syntax and semantics. Only some tables support updates. These tables are require to be at minimum

Reviewer: ```suggestion
Only some tables support updates. These tables must be at minimum
```

> format version 2 and the ``write.update.mode`` must be set to `merge-on-read`

Reviewer: ```suggestion
format version 2, and the ``write.update.mode`` must be set to `merge-on-read`.
```

.. code-block:: sql

    UPDATE region SET name = 'EU', comment = 'Europe' WHERE regionkey = 1;

.. code-block:: text

    UPDATE: 1 row

    Query 20250204_010341_00021_ymwi5, FINISHED, 2 nodes

> The query will error if the table does not meet the requirements for updates.

Reviewer: ```suggestion
The query returns an error if the table does not meet the requirements for updates.
```
